doc/user_guide.md

```diff
@@ -0,0 +1,435 @@
+     ...         '-- java          unit and integration tests
+     
+
+In this case the minimal OFT command line looks like this:
+
+```sh
+PROJECT_ROOT='/home/git/my-project/'
+oft trace "$PROJECT_ROOT"/doc "$PROJECT_ROOT"/src/main/java "$PROJECT_ROOT"/src/test/java
+```
+
+Or if you prefer it shorter:
+
+```sh
+cd /home/git/my-project/
+oft trace doc src/main/java src/test/java
+```
+
+The first variant is better suited for integration into scripts where you usually want to avoid changing the directory.
+
+By default this will produce a plain text trace that displays a details of all defect specification items and a summary.
+
+See also:
+* [Tracing Options](#tracing-options) for controlling the tracing output
+
+## Reference
+
+### OFT Command Line
+
+The OFT command line looks like this:
+
+    oft command [option ...] [<input file or directory> ...]
+
+Where `command` is one of
+
+* `trace` - create a requirement trace document
+* `convert` - convert to a different requirements format
+
+and `option` is one or more of the options listed below.
+
+#### Tracing options
+
+    -o, --output-format <format>
+
+The format of the report.
+
+One of:
+* `plain`
+
+Defaults to `plain`.
+
+    --v, --verbosity-level <level>
+
+The verbosity of the tracing report.
+
+* `quiet` - no output (in case only the return code is used)
+* `minimal` - display `ok` or `not ok`
+* `summary` - display only the summary, not individual specification items
+* `failures` - list of defect specification items
+* `failure_summaries` - list of summaries for defect specification items
+* `failure_details` - summaries and details for defect specification items
+* `all` -  summaries and details for all specification items
+
+Defaults to `failure_details`.
+
+#### Converting Options
+
+    -o, --output-format <format>
+
+Format into which requirements are converted.
+
+One of
+* `specobject`
+
+Defaults to `specobject`.
+
+#### Common Options
+
+    -f, --file <path>
+
+The output file or in case the output consists of more than one file, the output path. Defaults to STDOUT if not given.
+
+    -i, --ignore-artifact-types <type<[,<type> ...]
+
+Choose one or more artifact types which are going to be ignored during import. Affects specification items of that type, needed coverage and links to specification items of that type.
+
+    -n, --newline <format>
+
+Newline format, one of
+* `unix`
+* `windows`
+* `oldmac`
+
+Defaults to the platform standard if not given.
+
+## OFT API
+
+If you are a software developer planning to integrate OFT into one of your programs or scripts, you will probably want to use the OFT API.
+
+Below you find a few short examples of how to use the OFT API. For details check the JavaDoc documentation of the interfaces in the source code.
+
+* [org.itsallcode.openfasttrace.Converter](../src/main/java/org/itsallcoded/openfasttrace/Converter.java)
+* [org.itsallcode.openfasttrace.Reporter](../src/main/java/org/itsallcoded/openfasttrace/Reporter.java)
```

Comment:
Links don't work when viewing the file on a branch. Generated broken link: `https://github.com/itsallcode/openfasttrace/blob/story/%23111-Add_handbook/src/main/java/org/itsallcoded/openfasttrace/Converter.java`